qml/MainScreen.qml

```diff
@@ -217,6 +227,56 @@ Item {
         defaultValue: "file:///usr/share/asteroid-launcher/applauncher/000-default-horizontal.qml"
     }
 
+    ConfigurationValue {
+        id: watchfaceNightstandSource
+        key: "/desktop/asteroid/nightstand/watchface"
+        defaultValue: "file:///usr/share/asteroid-launcher/watchfaces/005-analog-nordic.qml"
+    }
+
+    ConfigurationValue {
+        id: nightstandBrightness
+        key: "/desktop/asteroid/nightstand/brightness"
+        defaultValue: 30
+    }
+
+    ConfigurationValue {
+        id: nightstandDelay
+        key: "/desktop/asteroid/nightstand/delay"
+        defaultValue: 5
+    }
+
+    ConfigurationValue {
+        id: nightstandEnabled
+        key: "/desktop/asteroid/nightstand/enabled"
+        defaultValue: false
+    }
+
+    Item {
+        id: nightstandMode
+        readonly property bool active: ready || (nightstandEnabled.value && holdoff)
```

Comment:
Am I right that if you remove `ready || (nightstandEnabled.value &&` here the behavior would not change?

There shouldn't be any situation when `ready` is true and `holdoff` is false because `holdoff` gets set to true everytime `ready` is set to true and it only transitions to false a while after `ready` goes back to false. So the `ready ||` part seems redundant ?

Also there shouldn't be any situation when `holdoff` is true and `nightstandEnabled.value` is false because `holdoff` is only set to true when ready transitions to true which can only happens if `nightstandEnabled.value` is true...

So how about removing this definition of `ready` and the holdoff variable and write to ready  instead of write to holdoff ?

Reply:
The logic was a bit convoluted, but now it implements this state machine using only two boolean properties.  New code comments also should help explain.
![transitions](https://user-images.githubusercontent.com/4911249/191634576-b9c516c8-9a2d-4489-93d0-da3555f7110e.svg)